middleware/README

```diff
@@ -13,10 +13,10 @@ the response from the HTTP handler it wraps, compute a checksum, add the
 checksum as an HTTP header called `X-Checksum`, and send the response.
```

Comment:
Hi,

I'm still figuring out proper pull request etiquette, and it now occurs to me that I should have included some context along these lines:

I have really been enjoying working through these exercises, and in particular the 'go' one since it's my first time working with the language. I spotted a couple spots in the README that might benefit from some adjustment, so that's what I'm submitting here. I realize some of these changes are more subjective than others, so if you'd like me to make modifications to the PR, I'm glad to do that. Thanks again for making these exercises available.